fcrepo-kernel/src/main/java/org/fcrepo/kernel/exception/ConstraintViolationException.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.exception;
+
+import static org.fcrepo.kernel.RdfLexicon.CONSTRAINED_BY;
+
+import javax.ws.rs.core.Link;
+import javax.ws.rs.core.UriInfo;
+
+/**
+ * A constraint has been violated.
+ *
+ * @author whikloj
+ * @since 2015-05-29
+ */
+public class ConstraintViolationException extends MalformedRdfException {
```

Comment:
Most of these exceptions are due to errors in the RDF passed to Fedora. It is possible that the error is because the client was unaware of our constraint, but it is an error in the RDF as we see it.

It also made it easier to fit into the existing architecture as there are some ```try {} catch {}``` that handle ```MalformedRdfException``` that did not therefore need to be changed.

It can however be made a direct subclass of ```RepositoryRuntimeException``` if you prefer.